src/components/content/order/ServiceList.tsx

```diff
@@ -0,0 +1,210 @@
+
+    function sortVersion(a: ServiceVo, b: ServiceVo) {
+        const aParts = a.version.split('.').map((part) => parseInt(part, 10));
+        const bParts = b.version.split('.').map((part) => parseInt(part, 10));
+        for (let i = 0; i < Math.max(aParts.length, bParts.length); i++) {
+            if (aParts[i] === undefined) {
+                return -1; // a is shorter, so it should come first
+            }
+            if (bParts[i] === undefined) {
+                return 1; // b is shorter, so it should come first
+            }
+            if (aParts[i] < bParts[i]) {
+                return -1; // a should come before b
+            }
+            if (aParts[i] > bParts[i]) {
+                return 1; // b should come before a
+            }
+        }
+        return 0; // a and b are equal
+    }
+
+    const columns: ColumnsType<ServiceVo> = [
+        {
+            title: 'Id',
+            dataIndex: 'id',
+        },
+        {
+            title: 'Category',
+            dataIndex: 'category',
+            filters: categoryFilters,
+            filterMode: 'tree',
+            filterSearch: true,
+            onFilter: (value: string | number | boolean, record) => record.category.startsWith(value.toString()),
+        },
+        {
+            title: 'Name',
+            dataIndex: 'name',
+            filters: nameFilters,
+            filterMode: 'tree',
+            filterSearch: true,
+            onFilter: (value: string | number | boolean, record) => record.name.startsWith(value.toString()),
+        },
+        {
+            title: 'Version',
+            dataIndex: 'version',
+            filters: versionFilters,
+            filterMode: 'tree',
+            filterSearch: true,
+            onFilter: (value: string | number | boolean, record) => record.version.startsWith(value.toString()),
+            sorter: (service1, service2) => sortVersion(service1, service2),
+        },
+        {
+            title: 'Csp',
+            dataIndex: 'csp',
+            filters: cspFilters,
+            filterMode: 'tree',
+            filterSearch: true,
+            onFilter: (value: string | number | boolean, record) => record.csp.startsWith(value.toString()),
+        },
+        {
+            title: 'Flavor',
+            dataIndex: 'flavor',
+        },
+        {
+            title: 'ServiceState',
+            dataIndex: 'serviceState',
+        },
+        {
+            title: 'Operation',
+            dataIndex: 'operation',
+            render: (text, record) => (
+                <Space size='middle'>
+                    <Button type='primary' icon={<CopyOutlined />}>
+                        migrate
+                    </Button>
+                    <Button type='primary' icon={<CloseCircleOutlined />}>
+                        destroy
+                    </Button>
+                </Space>
+            ),
+        },
+    ];
+
+    function getCspFilters(resp: ServiceVo[]) {
```

Comment:
all the `get` methods are actually updating states. So better to rename the method with word `update`

Reply:
fixed.